module/src/main/java/zipkin/module/aws/elasticsearch/AWSSignatureVersion4.java

```diff
@@ -84,11 +85,13 @@ static Function<HttpClient, HttpClient> newDecorator(String region,
   @Override public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) {
     // We aggregate the request body with pooled objects because signing implies reading it before
     // sending it to Elasticsearch.
-    return HttpResponse.from(
-        req.aggregateWithPooledObjects(ctx.eventLoop(), ctx.alloc()).thenApply(aggReg -> {
+    return HttpResponse.of(
+        req.aggregate(AggregationOptions.usePooledObjects(ctx.alloc())).thenApply(aggReg -> {
           try {
             AggregatedHttpRequest signed = sign(ctx, aggReg);
-            return unwrap().execute(ctx, signed.toHttpRequest());
+            final HttpRequest httpRequest = signed.toHttpRequest();
```

Comment:
not terrible drift considering the years passed!

Reply:
yeah! kudos to you man!